src/cascadia/TerminalSettingsEditor/Actions.cpp

```diff
@@ -117,16 +129,33 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     {
         _State = e.Parameter().as<Editor::ActionsPageNavigationState>();
 
+        // Populate AvailableActionAndArgs
+        std::map<hstring, Editor::ActionAndArgsViewModel> availableActionVMMap;

```

Comment:
Might be a dumb question - would it be enough to rely on this map of string->action so that you can just have an observable vector of strings, and proposed/current action could also just be the name, and if you need to update the state u can just lookup in the map?